src/Groups/pcgroup.jl

```diff
@@ -0,0 +1,345 @@
+    if GAP.Globals.IsSingleCollectorRep(c.X)
+      # For this type of collectors, `GAP.Globals.ShallowCopy`
+      # returns an independent copy.
+      cc.X = GAP.Globals.ShallowCopy(c.X)
+    elseif GAP.Globals.IsFromTheLeftCollectorRep(c.X)
+      # Currently there is no `GAP.Globals.ShallowCopy` method for `c.X`.
+      # Create the GAP object anew.
+      cc.X = _GAP_collector_from_the_left(c)
+    else
+      error("unknown GAP collector")
+    end
+  end
+
+  return cc
+end
+
+# Create a new GAP collector using `GAP.Globals.SingleCollector`.
+function _GAP_single_collector(c::GAP_Collector)
+  G = free_group(c.ngens; eltype = :syllable)
+
+  cGAP = GAP.Globals.SingleCollector(G.X, GapObj(c.relorders, recursive = true))::GapObj
+  for i in 1:c.ngens
+    GAP.Globals.SetPower(cGAP, i, G(c.powers[i]).X)
+  end
+  for j in 1:c.ngens
+    for i in 1:(j-1)
+      if isassigned(c.conjugates, i, j)
+        GAP.Globals.SetConjugate(cGAP, j, i, G(c.conjugates[i, j]).X)
+      else
+        GAP.Globals.SetConjugate(cGAP, j, i, G([j => 1]).X)
+      end
+    end
+  end
+  GAP.Globals.UpdatePolycyclicCollector(cGAP)
+
+  return cGAP, G
+end
+
+
+# Create a new GAP collector using `GAP.Globals.FromTheLeftCollector`
+# (from GAP's Polycyclic package).
+function _GAP_collector_from_the_left(c::GAP_Collector)
+  cGAP = GAP.Globals.FromTheLeftCollector(c.ngens)
+  for i in 1:c.ngens
+    if c.relorders[i] != 0
+      # We are not allowed to set relative orders to zero.
+      GAP.Globals.SetRelativeOrder(cGAP, i, GAP.Obj(c.relorders[i]))
+    end
+  end
+  for i in 1:c.ngens
+    if c.relorders[i] != 0
+      GAP.Globals.SetPower(cGAP, i, _GAP_generator_exponent_vector(c.powers[i]))
+    end
+  end
+  for j in 1:c.ngens
+    for i in 1:(j-1)
+      if isassigned(c.conjugates, i, j)
+        GAP.Globals.SetConjugate(cGAP, j, i, _GAP_generator_exponent_vector(c.conjugates[i, j]))
+      else
+        GAP.Globals.SetConjugate(cGAP, j, i, GapObj([j, 1]))
+      end
+    end
+  end
+  GAP.Globals.UpdatePolycyclicCollector(cGAP)
+  
+  return cGAP
+end
+
+
+# Create the collector on the GAP side on demand
+function Base.getproperty(c::GAP_Collector, sym::Symbol)
+  isdefined(c, sym) && return getfield(c, sym)
+  if sym === :X
+    if 0 in c.relorders
+      # create a collector from the Polycyclic package.
+      cGAP = _GAP_collector_from_the_left(c)
+    else
+      # create a GAP library collector for finite polycyclic groups.
+      cGAP, F = _GAP_single_collector(c)
+      c.F = F
+    end
+    c.X = cGAP
+  end
+  return getfield(c, sym)
+end
+
+"""
+    pc_group(c::GAP_Collector)
+
+Create a `PcGroup` object from `c`.
+
+# Examples
+```jldoctest
+julia> c = collector(2, Int);
+
+julia> Oscar.set_relative_orders!(c, [2, 3])
+
+julia> Oscar.set_conjugate!(c, 2, 1, [2 => 2])
+
+julia> gg = pc_group(c)
+<pc group of size 6 with 2 generators>
+
+julia> describe(gg)
+"S3"
+```
+"""
+function pc_group(c::GAP_Collector)
+  if 0 in c.relorders
```

Comment:
Could we perhaps instead check the filters of `c.X` to decide which way we go? Then in the future we could offer a way to optionally also create a pcp group even if the collector describes a finite group (I am not sure we *want* that, but at least it would be an option. Actually I can think of one advantage pcp groups have there: e.g. a cyclic group of order 10^6 can be described with one generator as a pcp group, but not as a pc group (at least not if one wants that pc group the useful)